Feat/timetable

byungchanKo99 opened the pull request:
## 결과
|첫화면|중간화면|모달|
|--|--|--|
|<img width="200" alt="스크린샷 2021-08-05 오전 3 08 47" src="https://user-images.githubusercontent.com/70642609/128232573-05fe40e5-d126-4642-84e3-9e99e0ef3e45.png">|<img width="200" alt="스크린샷 2021-08-05 오전 3 08 11" src="https://user-images.githubusercontent.com/70642609/128232735-f5f1ac63-b810-43b1-84a0-29f6d0f435dc.png">|<img width="200" alt="스크린샷 2021-08-05 오전 3 08 25" src="https://user-images.githubusercontent.com/70642609/128232753-58d7e88a-b845-4260-9354-9419d40f6edf.png">|
## 구조
`dayColumns` : 요일 라벨(day) 감싸는 div.
`timeIndex` : 시간라벨(time) 감싸는 div. 이때 각 열마다 오른쪽 빈 박스도 같이 출력함.
`times`, `week`, `schedule` : 시간, 요일, 일정 담고 있는 변수리스트. -> 통신할 때는 state로 바꿔야 하지 않을까용
`reserveBox` : 예약상황 보여주는 박스. position absolute로 위치시킴. schedule 리스트가 이중으로 쌓여 있어서 이중 map써서 출력함. 서로 다른 위치 스타일을 줘야 해서 인라인으로 스타일 주었음.

`colorGenerator` : schedule인덱스 활용해서 색상 부여함.
`xPosGenerator`: left position 부여함. `schedule`리스트에서 최상위..? 인덱스 써서 월(0) 화(1) 수(2) 목(3) 금(4).... 이런식으로 값 준 뒤 * width 해줌.
`yPosGenerator`: top position 부여함. 예약정보 중 시작시간에서 시간부분 슬라이스 해서 * height 해줌. 분 단위는 분 부분 슬라이스 해서 30분으로 나누고 * (height /2) 해줌.
`heightGenerator` : 박스 높이 정해줌. 시간 차이는 예약할 때 최대 한시간으로 정해져서 분만 비교함. |시작 분 - 끝 분| 해서 00이면 1시간, 30이면 30분으로 생각함.


## 문제
1. 최상위 `div`에 marginBottom을 안넣으면 시간표가 끝까지 스크롤되지 않음. -> 이유 모르겠음....
2. 모달창 텍스트 형식이 달라서 이름 외 내용 출력할 때 글씨가 너무 작음.

byungchanKo99 committed:
feat: 시간, 요일 리스트 생성

byungchanKo99 committed:
feat: 요일 열 구현

byungchanKo99 committed:
feat: 시간 행 구현

byungchanKo99 committed:
feat: 내부 빈칸 구현

byungchanKo99 committed:
feat: 시간표 파일 분리

byungchanKo99 committed:
feat: 더미데이터

데이터 리스트 임의로 넣은 상태.

byungchanKo99 committed:
feat: 예약박스 &  xy좌표 및 높이 생성함수

byungchanKo99 committed:
feat: 예약박스 색 생성 함수

byungchanKo99 committed:
feat: 예약박스 출력

byungchanKo99 committed:
feat: 박스 클릭시 모달 출력

byungchanKo99 committed:
fix: 모달 버튼 텍스트 변경

Aperture507 committed:
build: pod install

DonghyunKim98 commented on app/components/screens/reservation/index.tsx:
```diff
@@ -7,14 +7,48 @@ import {
+    width: widthPercentage(46),
+    alignItems: 'center',
+    borderLeftWidth: 0,
+    borderWidth: 1,
```
- 이런 식이면 뒤에 선언된 것이 우선순위가 높기 때문에 왼쪽이 제외되지 않고 left/right/top/bottom 에 다 border 이 생깁니다.
- 반응형 작업해주세요.

DonghyunKim98 commented on app/components/screens/reservation/index.tsx:
```diff
@@ -49,6 +49,22 @@ const styles = StyleSheet.create({
+    height: heightPercentage(46),
+    alignItems: 'flex-end',
+    borderTopWidth: 0,
+    borderWidth: 1,
```
- 같은 리뷰입니다.

DonghyunKim98 commented on app/components/screens/reservation/index.tsx:
```diff
@@ -7,70 +7,22 @@ import {
+//   heightPercentage,
+//   widthPercentage,
+//   fontPercentage,
+// } from '../../../utils/constant/common/design/Responsive';
```
- 주석처리를 하지말고 삭제를 해주세요.

DonghyunKim98 commented on app/components/screens/reservation/index.tsx:
```diff
@@ -102,28 +42,12 @@ function Reservation() {
+          <ScrollView>
+            <Text>버튼 있는 자리</Text>
+            <TimeTable />
+          </ScrollView>
```
- 스크롤 뷰에 애를 먹고 있다면, style 쪽 문제일 가능성이 큽니다.
- `contentContainerStyle` 을 활용해보세요!

DonghyunKim98 commented on app/components/screens/reservation/timeTable.tsx:
```diff
@@ -73,6 +73,43 @@ function TimeTable() {
   const times = generateTimes(0, 24);
   const week = ['MON', 'TUE', 'WEN', 'THU', 'FRI', 'SAT', 'SUN'];
 
+  const schedule = [
```
- var naming 을 tempSchedule 로 주면 좋을 것 같습니다.

DonghyunKim98 commented on app/components/screens/reservation/timeTable.tsx:
```diff
@@ -110,6 +116,23 @@ function TimeTable() {
+    return Math.abs(parseInt(start.slice(3), 10) - parseInt(end.slice(3), 10))
+      ? heightPercentage(23)
+      : heightPercentage(46);
+  }
```
```tsx
 function heightGenerator(start: string, end: string) {
   const defaultHeight = heightPercentage(23);
    return Math.abs(parseInt(start.slice(3), 10) - parseInt(end.slice(3), 10))
      ? defaultHeight
      : defaultHeight * 2 ;
  }
```
- 제가 준 코드 처럼 의미가 있게 작성해주세요.
- return 문에 있는 조건문도 사실 무얼 의미하는지 알 수 가 없습니다. 😅 상수화 시켜주세요.

DonghyunKim98 commented on app/components/screens/reservation/timeTable.tsx:
```diff
@@ -110,6 +116,23 @@ function TimeTable() {
     [], // 토
     [], // 일
   ];
+  function xPosGenerator(day: number): number {
```
- React Function 안에 작성되는 함수는 Arrow Function 으로 나타내는게 컨벤션입니다.

byungchanKo99 committed:
feat: border 반응형 위치 수정

byungchanKo99 committed:
feat: 주석 삭제

byungchanKo99 committed:
feat: schedule 변수 이름 수정

byungchanKo99 committed:
feat: 내부 함수 arrow function으로 변경

byungchanKo99 committed:
feat: 내부 메소드 리턴값 표기 수정

byungchanKo99 committed:
feat: screenWrapper 코드 붙이기

customHeader에서 title이 undefined로 초기화 되는 경우 Type 처리

byungchanKo99 committed:
feat: screenWrapper적용

screenWrapper 적용하니까 스크롤 문제 해결됨

byungchanKo99 committed:
Merge branch 'feat/timetable' of https://github.com/Hanwoollim-App/Hanwoollim-managers into feat/timetable

DonghyunKim98 commented on app/utils/constant/reservation/index.ts:
```diff
@@ -0,0 +1,7 @@
+  ColumnsHeight: 14,
+  IndexWidth: 20,
+  defaultBoxSize: 46,
+};
```
- 반응형으로 작업할수는 없는건가요?

DonghyunKim98 commented on app/components/common/CustomHeader.tsx:
```diff
@@ -38,7 +38,7 @@ const styles = StyleSheet.create({
 
 interface HeaderInterface {
-  headerTitle: String;
+  headerTitle: String | undefined;
```
- undefined 보다는 `?` 를 이용해서 optional props 로 나타내는게 더 좋을 것 같습니다.

byungchanKo99 committed:
feat: headerTitle optional로 변경

This reverts commit 03c3ee94f95caca2610b3f96bed4c0105f97ac42, reversing
changes made to 86561f846ca40c7647887d16556ca6432f2b21b4.

byungchanKo99 committed:
feat: boxSize 반응형 적용

byungchanKo99 committed:
feat: JSX is not defined 에러

dropDown.ts에서 JSX is not defined - eslint  에러가 떠서  찾아보니까 ts에서 자체적으로 nodef를 체크해서 eslint 옵션은 꺼도 된다합니당...

byungchanKo99 committed:
feat: 예약화면 UI 적용

byungchanKo99 committed:
feat: value state 초기값 수정

ValueType에 null을 할당할 수 없다고 떠서 items의 0번 인덱스를 줬습니다. 그러니 items[0].value가 undefined일 수 있어 할당할 수 없다는 에러가 떠서 ItemType에서 value을 필수로 바꿔봤는데 괜찮을까용

byungchanKo99 committed:
Merge branch 'dev' into feat/timetable

DonghyunKim98 commented on app/components/screens/reservation/index.tsx:
```diff
@@ -97,13 +97,14 @@ function ReservationTimeTable() {
     {label: '7.11~7.18', value: '7.11~7.18'},
     {label: '7.18~7.25', value: '7.18~7.25'},
   ]);
+  const [value, setValue] = useState<ValueType>(items[0].value);
```
```
const [value, setValue] = useState<ValueType|null>(null);
```
- 이런식으로 가는게 더 괜찮을 것 같습니다.

byungchanKo99 committed:
feat: value type 수정

byungchanKo99 committed:
Merge branch 'feat/timetable' of https://github.com/Hanwoollim-App/Hanwoollim-managers into feat/timetable

byungchanKo99 committed:
fix: lint error 해결
